### PR TITLE
Add qe-test-coverage for BZ-1900575, 1905801 and 1886031

### PR DESCRIPTION
**Pull request by Gauravtalreja1**
**Test Results:**
```
$ pytest -sv --junit-xml=foreman-results.xml --ansible-host-pattern server --ansible-user root --ansible-inventory ../testfm/inventory test_health.py -k test_positive_available_space_candlepin
========================================================== test session starts ==========================================================
platform linux -- Python 3.7.7, pytest-3.6.1, py-1.10.0, pluggy-0.6.0 -- /home/gtalreja/sat_workspace/testfm/.testfm/bin/python
cachedir: ../.pytest_cache
ansible: 2.6.19
rootdir: /home/gtalreja/sat_workspace/testfm, inifile:
plugins: ansible-2.2.2
collected 25 items / 24 deselected                                                                                                      

test_health.py::test_positive_available_space_candlepin 2021-03-15 15:24:51,505 - testfm.log - INFO - Running ForemanMaintain::Scenario::FilteredScenario
================================================================================
Check to make sure /var/lib/candlepin has enough space:               [OK]
--------------------------------------------------------------------------------
PASSED

--------------------------- generated xml file: /home/gtalreja/sat_workspace/testfm/tests/foreman-results.xml ---------------------------
=============================================== 1 passed, 24 deselected in 18.16 seconds ================================================
```
```
$ pytest -sqv --junit-xml=foreman-results.xml --ansible-host-pattern server --ansible-user root --ansible-inventory testfm/inventory tests/test_service.py -k test_positive_foreman_maintain_service_list_sidekiq
========================================================== test session starts ==========================================================
platform linux -- Python 3.7.7, pytest-3.6.1, py-1.10.0, pluggy-0.6.0
ansible: 2.6.19
rootdir: /home/gtalreja/sat_workspace/testfm, inifile:
plugins: ansible-2.2.2
collected 11 items / 10 deselected                                                                                                      

tests/test_service.py 2021-03-15 15:29:55,264 - testfm.log - INFO - Running Service List
================================================================================
List applicable services: 
dynflow-sidekiq@.service                      enabled 
foreman-proxy.service                         enabled 
httpd.service                                 enabled 
postgresql.service                            enabled 
pulp_celerybeat.service                       enabled 
pulp_resource_manager.service                 enabled 
pulp_streamer.service                         enabled 
pulp_workers.service                          enabled 
puppetserver.service                          enabled 
qdrouterd.service                             enabled 
qpidd.service                                 enabled 
rh-mongodb34-mongod.service                   enabled 
rh-redis5-redis.service                       enabled 
smart_proxy_dynflow_core.service              enabled 
squid.service                                 enabled 
tomcat.service                                enabled

All services listed                                                   [OK]
--------------------------------------------------------------------------------
2021-03-15 15:32:33,402 - testfm.log - INFO - Running Restart Services
================================================================================
Check if command is run as root user:                                 [OK]
--------------------------------------------------------------------------------
Restart applicable services: 

Stopping the following service(s):
rh-mongodb34-mongod, postgresql, qdrouterd, qpidd, rh-redis5-redis, squid, pulp_celerybeat, pulp_resource_manager, pulp_streamer, pulp_workers, smart_proxy_dynflow_core, tomcat, dynflow-sidekiq@orchestrator, httpd, puppetserver, dynflow-sidekiq@worker, dynflow-sidekiq@worker-hosts-queue, foreman-proxy
/ All services stopped                                                          

Starting the following service(s):
rh-mongodb34-mongod, postgresql, qdrouterd, qpidd, rh-redis5-redis, squid, pulp_celerybeat, pulp_resource_manager, pulp_streamer, pulp_workers, smart_proxy_dynflow_core, tomcat, dynflow-sidekiq@orchestrator, httpd, puppetserver, dynflow-sidekiq@worker, dynflow-sidekiq@worker-hosts-queue, foreman-proxy
- All services started                                                [OK]      
--------------------------------------------------------------------------------
.

------------------------------ generated xml file: /home/gtalreja/sat_workspace/testfm/foreman-results.xml ------------------------------
=============================================== 1 passed, 10 deselected in 168.30 seconds ===============================================
```

Signed-off-by: Gaurav Talreja <gtalreja@redhat.com>

### CODE REVIEW

File: tests/test_upgrade.py [hunk 1]
```diff
@@ -124,3 +127,30 @@ def test_positive_self_update():
...
+
+        :steps:
+            1. Run foreman-maintain upgrade list-versions/check/run command.
+            2. Run foreman-maintain upgrade list-versions/check/run command,
+                after removing satellite and satellite-capsule packages.
+
+        :BZ: 1886031
+
+        :expectedresults:
+            2. If those packages are removed, then it should give error, like
```
Comment (jameerpathan111) [inline tests/test_upgrade.py:151]:
```suggestion
            1. If those packages are removed, then it should give error, like
```

**Review by jameerpathan111 — APPROVED** [review]
ack

**Review by lvrtelov — COMMENTED** [review]
ACK

**Review by vsedmik — APPROVED** [review]
LGTM :+1:

**Pull request merged**